hera_pspec/pspecdata.py

```diff
@@ -2023,14 +2182,24 @@ def scalar_delay_adjustment(self, key1=None, key2=None, sampling=False,
         ratio[np.isnan(ratio)] = 1.0
         ratio[np.isinf(ratio)] = 1.0
 
-        # get mean ratio
-        mean_ratio = np.mean(ratio)
-        scatter = np.abs(ratio - mean_ratio)
-        if (scatter > 10**-4 * mean_ratio).any():
-            raise ValueError("The normalization scalar is band-dependent!")
-
-        adjustment = self.spw_Ndlys / (self.spw_Nfreqs * mean_ratio)
-
+        ## XXX: Adjustments like this are hacky and wouldn't be necessary
+        ## if we deprecate the incorrectly normalized
+        ## Q and M matrix definitions.
+        #In the future, we need to do our normalizations properly and
+        #stop introducing arbitrary normalization factors.
+        #if the input identity weighting is diagonal, then the
+        #adjustment factor is independent of alpha.
+        # get mean ratio.
+        if self.data_weighting == 'identity':
+            mean_ratio = np.mean(ratio)
+            scatter = np.abs(ratio - mean_ratio)
+            print(scatter)
+            if (scatter > 10**-4 * mean_ratio).any():
+                raise ValueError("The normalization scalar is band-dependent!")
+            adjustment = self.spw_Ndlys / (self.spw_Nfreqs * mean_ratio)
+        #otherwise, the adjustment factor is dependent on alpha.
+        else:
+            adjustment = self.spw_Ndlys / (self.spw_Nfreqs * ratio)
```

Comment:
I do not understand the details of this change. May be @acliu can comment?

Reply:
@acliu , can you also review this bit?

Reply:
This looks good, and is the right way to compute the adjustment. I just want to make a few requests:

- The docstring for this function says that it returns a float (rather than an array of floats), but this is not true for the arbitrary weights/tapers.
- In the test code for this function, there's a test to make sure an array of the right length is returned if there is a non-trivial taper. Let's extend this test to also encompass this new case.
- The phrase "In general, the result is still independent of alpha" is still in the docstring. Let's qualify this.
- Embarrassingly, I don't remember why there is a factor of ``self.spw_Ndlys / self.spw_Nfreqs`` in the adjustment. I'm sure it's right (since the original version passes all the validation tests), but can you remind me where this comes from?

Reply:
* I've updated the docstring to note that `adjustment` is only a float for `identity` weighting. 
* I've added the length check to the `test_scalar_delay_adjustment` unittest. According to my notes, the norm factor is independent of alpha as long as the R matrix is diagonal (so diagonal non-trivial tapers are OK with a scalar norm factor). 
* I've changed the docstring so that it mentions that if data_weighting is equal to identity, then there is a single float normalization and if not, we generally need a different adjustment for each alpha. 
*I think the Ndlys / Nfreqs comes out of equation 47 and 48 from your memo that you shared with me privately. I'd be happy to post it on this thread for posterity but leave it up to you. For some reason doesn't appear to be on reionization.org. Memo 43 is close but it doesn't have anything on the adjustments Maybe ask @dannyjacobs  to add it?

Reply:
Ok, this is all fine. I'll ask Danny to update the memo. Thanks for your patience on all this!